internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go

```diff
@@ -117,3 +186,10 @@ func labelsToDims(labels []*dto.LabelPair) map[string]string {
 	}
 	return dims
 }
+
+func applyDimensions(attrs pcommon.Map, extraDimensions map[string]string) {
```

Comment:
Can this be moved to the output method like is done for `SendDatapoints` to avoid the requirement of adding to all monitors? I think adding the missing filtering pass there is also still required unless I'm missing it.

Reply:
Yes, I'll do that, that should be easier to repurpose indeed.

Reply:
Done.

Reply:
To clarify I'm requesting that the existing functionality for the sfx data be added to your new `SendMetrics` method for compatibility and to avoid having to add the logic in monitors: https://github.com/signalfx/splunk-otel-collector/blob/main/pkg/receiver/smartagentreceiver/output.go#L205

Reply:
Is the PR now in line with your request above? Just making sure I am understanding correctly.